### PR TITLE
Toolchain: Add support for Clang on MacOS

### PR DESCRIPTION
**Pull request by jepebe**
This commit adds Darwin as a possible host for building the toolchain with Clang. It requires GNU sed and Coreutils to be able to build.

### CODE REVIEW

**Comment by ADKaster** [comment]
Could you try your BuildClang.sh changes along with my patches from #11584 ? Also, looks like `shellcheck` wants you to quote "$INSTALL"  where it's used. You can setup the pre-commit hooks using [`pre-commit`](https://github.com/SerenityOS/serenity/blob/0306cf203062355c646fff1248108d839346e018/CONTRIBUTING.md#commit-hooks) if you wanna catch those earlier :).

**Review by BertalanD — APPROVED** [review]
LGTM!

File: Toolchain/CMake/LLVMRuntimesConfig.cmake [hunk 1]
```diff
@@ -11,6 +11,10 @@ set(LLVM_ENABLE_RUNTIMES "libcxx;libcxxabi;libunwind" CACHE STRING "")
 set(CMAKE_SYSTEM_NAME SerenityOS CACHE STRING "")
 set(target_triple ${SERENITY_TOOLCHAIN_ARCH}-pc-serenity)
 
+IF(${CMAKE_HOST_SYSTEM_NAME} MATCHES "Darwin")
+    set(CMAKE_BUILD_WITH_INSTALL_RPATH ON CACHE BOOL "")
+ENDIF()
```
Comment (trflynn89) [inline Toolchain/CMake/LLVMRuntimesConfig.cmake:16]: A bit of a nitpick, please use lowercase `if` and `endif`, and put a space after `if`:
```cmake
if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Darwin")
    set(CMAKE_BUILD_WITH_INSTALL_RPATH ON CACHE BOOL "")
endif()
```

Comment (trflynn89) [reply Toolchain/CMake/LLVMRuntimesConfig.cmake:16]: Oops, commented too late

Comment (alimpfard) [reply Toolchain/CMake/LLVMRuntimesConfig.cmake:16]: oops, didn't notice that ^^'

**Pull request merged**